feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,225 @@
+	subcomponents := gnmi.LookupAll[*oc.Component_Subcomponent](t, dut1, component1.SubcomponentAny().State())
+	for _, s := range subcomponents {
+		subc, ok := s.Val()
+		if ok {
+			sensorComponent := gnmi.Get[*oc.Component](t, dut1, gnmi.OC().Component(subc.GetName()).State())
+			if sensorComponent.GetType() == sensorType {
+				scomponent := gnmi.OC().Component(sensorComponent.GetName())
+				if scomponent != nil {
+					component1 = scomponent
+				}
+			}
+		}
+	}
+	p1StreamInstant := samplestream.New(t, dut1, component1.Temperature().Instant().State(), 10*time.Second)
+	p1StreamAvg := samplestream.New(t, dut1, component1.Temperature().Avg().State(), 10*time.Second)
+	p1StreamMin := samplestream.New(t, dut1, component1.Temperature().Min().State(), 10*time.Second)
+	p1StreamMax := samplestream.New(t, dut1, component1.Temperature().Max().State(), 10*time.Second)
+	// Wait 120 sec cooling off period
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_DOWN)
+	temperatureInstant := verifyTemperatureSensorValue(t, dut1, p1StreamInstant, "Instant")
+	t.Logf("Port1 dut1 %s Instant Temperature: %v", dp1.Name(), temperatureInstant)
+	temperatureMax := verifyTemperatureSensorValue(t, dut1, p1StreamMax, "Max")
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	temperatureMin := verifyTemperatureSensorValue(t, dut1, p1StreamMin, "Min")
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	temperatureAvg := verifyTemperatureSensorValue(t, dut1, p1StreamAvg, "Avg")
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	i = d.GetOrCreateInterface(dp1.Name())
+	i.Enabled = ygot.Bool(true)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	// Enable interface
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp1.Name()).Config(), i)
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	temperatureInstant = verifyTemperatureSensorValue(t, dut1, p1StreamInstant, "Instant")
+	t.Logf("Port1 dut1 %s Instant Temperature: %v", dp1.Name(), temperatureInstant)
+	temperatureMax = verifyTemperatureSensorValue(t, dut1, p1StreamMax, "Max")
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	temperatureMin = verifyTemperatureSensorValue(t, dut1, p1StreamMin, "Min")
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	temperatureAvg = verifyTemperatureSensorValue(t, dut1, p1StreamAvg, "Avg")
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	if temperatureAvg >= temperatureMin && temperatureAvg <= temperatureMax {
+		t.Logf("The average is between the maximum and minimum values")
+	} else {
+		t.Fatalf("The average is not between the maximum and minimum values")
+	}
+}
+
+func opticalChannelFromPort(t *testing.T, dut *ondatra.DUTDevice, p *ondatra.Port) string {
```

Comment:
should this name be componentFromPort?
out of curiosity did you run it against one of the FNT testbeds? What component name was returned for 400ZR? Something like Ethernet25-Optical0?

Reply:
Corrected the name as suggested opticalChannelComponentFromPort

thanks